app/objects/c_operation.py

```diff
@@ -272,7 +272,7 @@ def _get_all_possible_abilities_by_agent(self):
         return {a.paw: {'all_abilities': self.adversary.atomic_ordering} for a in self.agents}
 
     def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors, agent_ran):
-        variables = re.findall(r'#{(.*?)}', self.decode_bytes(ability.test), flags=re.DOTALL)
+        variables = re.findall(r'#{(.*?)}', self.decode_bytes(ability.test), flags=re.DOTALL) if ability.test else []
```

Comment:
can variables exists in `ability.code` as well?  If so, maybe we should we try to extract those here as well?

Reply:
Right now no